azkaban-web-server/src/main/java/cloudflow/servlets/ExecutionServlet.java

```diff
@@ -0,0 +1,102 @@
+package cloudflow.servlets;
+
+import azkaban.server.session.Session;
+import azkaban.webapp.AzkabanWebServer;
+import azkaban.webapp.servlet.LoginAbstractAzkabanServlet;
+import cloudflow.error.CloudFlowException;
+import cloudflow.models.JobExecution;
+import cloudflow.services.ExecutionService;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static cloudflow.servlets.Constants.*;
+
+public class ExecutionServlet extends LoginAbstractAzkabanServlet {
+
+    private static final Pattern GET_JOB_EXECUTION_INFO_PATTERN = Pattern.compile(
+        "/executions/([0-9]+)/jobs/([0-9]+)/([a-zA-Z1-9:_-]+)"); // TODO: remove "/[a-zA-Z1-9:_-]+"
+
+    private static final Logger logger = LoggerFactory.getLogger(ExecutionServlet.class);
+    private ExecutionService executorService;
+
+
+    @Override
+    public void init(final ServletConfig config) throws ServletException {
+        super.init(config);
+        final AzkabanWebServer server = (AzkabanWebServer) getApplication();
+        this.executorService = server.getExecutionService();
+    }
+
+    @Override
+    protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Session session)
+        throws ServletException, IOException {
+
+        Matcher matcher = GET_JOB_EXECUTION_INFO_PATTERN.matcher(req.getRequestURI());
+        if(matcher.matches()) {
```

Comment:
With a strictly defined regex we may be limited in our ability to be specific about what part of the request `path` failed the validation. 
Recommend looking at method `UriTemplate::match()` that will directly give us the template variables on a path without having to define a explicit regex.